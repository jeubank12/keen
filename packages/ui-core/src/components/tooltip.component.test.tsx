import React from 'react';
import { mount } from 'enzyme';
import Tooltip from './tooltip.component';
import BulletList from './bulletlist.component';

const setup = (overProps: any = {}) => {
  const props = {
    ...overProps,
  };

  const wrapper = mount(<Tooltip {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Tooltip />', () => {
  it('should render provided text', () => {
    const children = 'text';
    const { wrapper } = setup({
      children,
    });

    expect(wrapper.text()).toEqual(children);
    expect(wrapper.props().children).toEqual(children);
  });

  it('should render bullet points', () => {
    const bulletList = [
      {
        color: 'red',
        value: 'red',
      },
      {
        color: 'blue',
        value: 'blue',
      },
      {
        color: 'green',
        value: 'green',
      },
      {
        color: 'black',
        value: 'black',
      },
    ];

    const { wrapper } = setup({
      children: <BulletList list={bulletList} />,
    });

    expect(wrapper.find('li').length).toEqual(bulletList.length);
  });

  it('should render `dark-mode`', () => {
    const mode = 'dark';
    const children = 'tooltip';
    const { wrapper } = setup({
      mode,
      children,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render box shadow', () => {
    const hasShadow = false;
    const children = 'tooltip';
    const { wrapper } = setup({
      hasShadow,
      children,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
