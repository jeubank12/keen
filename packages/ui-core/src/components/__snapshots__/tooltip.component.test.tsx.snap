// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should not render box shadow 1`] = `
<Tooltip
  arrowDirection="bottom"
  backgroundColor="#FFFFFF"
  borderRadius="0px"
  fontColor="#3A494D"
  hasShadow={false}
  mode="light"
>
  <ThemeProvider
    theme={
      Object {
        "arrowDirection": "bottom",
        "mode": "light",
      }
    }
  >
    <styled.div
      backgroundColor="#FFFFFF"
      borderRadius="0px"
      fontColor="#3A494D"
      hasShadow={false}
    >
      <StyledComponent
        backgroundColor="#FFFFFF"
        borderRadius="0px"
        fontColor="#3A494D"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "bmwhaX",
              "rules": Array [
                "
  padding: 10px 15px;
  background-color: ",
                [Function],
                ";
  border-radius: ",
                [Function],
                ";
  font-family: ",
                [Function],
                ";
  font-size: ",
                [Function],
                ";
  color: ",
                [Function],
                ";
  box-shadow: ",
                [Function],
                ";
  position: relative;
  ",
                [Function],
                "
  &::after {
    content: '';
    position: absolute;
    width: 0;
    height: 0;
    box-sizing: border-box;
    border: 4px solid ",
                [Function],
                ";
    transform-origin: 0 0;
    box-shadow: ",
                [Function],
                ";
    ",
                [Function],
                "
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hasShadow={false}
      >
        <div
          className="sc-bdVaJa bmwhaX"
        >
          tooltip
        </div>
      </StyledComponent>
    </styled.div>
  </ThemeProvider>
</Tooltip>
`;

exports[`<Tooltip /> should render \`dark-mode\` 1`] = `
<Tooltip
  arrowDirection="bottom"
  backgroundColor="#FFFFFF"
  borderRadius="0px"
  fontColor="#3A494D"
  hasShadow={true}
  mode="dark"
>
  <ThemeProvider
    theme={
      Object {
        "arrowDirection": "bottom",
        "mode": "dark",
      }
    }
  >
    <styled.div
      backgroundColor="#FFFFFF"
      borderRadius="0px"
      fontColor="#3A494D"
      hasShadow={true}
    >
      <StyledComponent
        backgroundColor="#FFFFFF"
        borderRadius="0px"
        fontColor="#3A494D"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "hfYHmZ",
              "rules": Array [
                "
  padding: 10px 15px;
  background-color: ",
                [Function],
                ";
  border-radius: ",
                [Function],
                ";
  font-family: ",
                [Function],
                ";
  font-size: ",
                [Function],
                ";
  color: ",
                [Function],
                ";
  box-shadow: ",
                [Function],
                ";
  position: relative;
  ",
                [Function],
                "
  &::after {
    content: '';
    position: absolute;
    width: 0;
    height: 0;
    box-sizing: border-box;
    border: 4px solid ",
                [Function],
                ";
    transform-origin: 0 0;
    box-shadow: ",
                [Function],
                ";
    ",
                [Function],
                "
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hasShadow={true}
      >
        <div
          className="sc-bdVaJa hfYHmZ"
        >
          tooltip
        </div>
      </StyledComponent>
    </styled.div>
  </ThemeProvider>
</Tooltip>
`;
